pygpudrive/env/env_torch.py

```diff
@@ -104,32 +99,51 @@ def _apply_actions(self, actions):
         else:
             raise ValueError(f"Invalid action shape: {actions.shape}")
 
-        # Feed the actual action values to gpudrive
-        if self.action_features == 'delta_local':
-            self.sim.delta_action_tensor().to_torch().copy_(action_value_tensor)
+        # Feed the action values to gpudrive
+        self._copy_actions_to_simulator(action_value_tensor)
+
+    def _copy_actions_to_simulator(self, actions):
+        """Copy the provived actions to the simulator."""
+        if (
+            self.config.dynamics_model == "classic"
+            or self.config.dynamics_model == "bicycle"
+        ):
+            # Action space: (acceleration, steering, heading)
+            self.sim.action_tensor().to_torch()[:, :, :3].copy_(actions)
+        elif self.config.dynamics_model == "delta_local":
+            # Action space: (dx, dy, dyaw)
+            self.sim.action_tensor().to_torch()[:, :, 3:6].copy_(actions)
+            # Action space: (pos, yaw, velocity)
+        elif self.config.dynamics_model == "bicycle":
+            self.sim.action_tensor().to_torch()[:, :, 6:].copy_(actions)
```

Comment:
Couple of things here. The indexing is a bit wrong. I am adding in the correct indexing. Basically, you always index from 0. But you index only the number of floats that correspond to the dynamics model.

Also, I think the code can be simplified by removing the last elif.

```suggestion
        if (
            self.config.dynamics_model == "classic"
            or self.config.dynamics_model == "bicycle"
        ):
            # Action space: (acceleration, steering, heading)
            self.sim.action_tensor().to_torch()[:, :, :3].copy_(actions)
        elif self.config.dynamics_model == "delta_local":
            # Action space: (dx, dy, dyaw)
            self.sim.action_tensor().to_torch()[:, :, :3].copy_(actions)
            # Action space: (pos, yaw, velocity)
```

Reply:
Thanks, updated!